modules/phase_field/include/auxkernels/TotalFreeEnergyBase.h

```diff
@@ -0,0 +1,36 @@
+#ifndef TOTALFREEENERGYBASE_H
+#define TOTALFREEENERGYBASE_H
+
+#include "AuxKernel.h"
+
+//Forward Declarations
+class TotalFreeEnergyBase;
+
+template<>
+InputParameters validParams<TotalFreeEnergyBase>();
+
+/**
+ * Total free energy (both the bulk and gradient parts), where the bulk free energy has been defined in a material and called f_name
+ */
+class TotalFreeEnergyBase : public AuxKernel
+{
+public:
+  TotalFreeEnergyBase(const std::string & name, InputParameters parameters);
+
+protected:
+  virtual Real computeValue() = 0;
```

Comment:
I don't see any issues with this change set. You can leave that there if you want.